examples/abi/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::schemars::JsonSchema;
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{near_abi, near_bindgen};
+
+#[derive(JsonSchema, Serialize, Deserialize)]
+pub struct Pair(u32, u32);
+
+#[derive(JsonSchema, Serialize, Deserialize)]
+pub struct DoublePair {
+    first: Pair,
+    second: Pair,
+}
+
+near_abi! {
```

Comment:
Did you end up experimenting with having this as an annotation for a module or including logic within `near_bindgen` somehow?

Reply:
Not yet, but it is in my plans

Reply:
So I have played with the idea and surprisingly everything seems to be working as is. I will run it through a few more scenarios later, in the meantime please take another look when you can.